src/randoop/ExceptionCheck.java

```diff
@@ -0,0 +1,145 @@
+package randoop;
+
+import java.io.ObjectStreamException;
+import java.lang.reflect.Modifier;
+import java.util.Objects;
+
+/**
+ * An {@code ExceptionCheck} is used to indicate that an exception is expected
+ * at a particular statement in a sequence. Depending on command line arguments
+ * to Randoop, an instance may be either a {@link ExpectedExceptionCheck} or 
+ * {@link EmptyExceptionCheck}. Both will catch the exception if it occurs, but
+ * differ on whether the expectation of the exception is enforced.
+ */
+public abstract class ExceptionCheck implements Check {
+  
+ /**
+   * 
+   */
+  private static final long serialVersionUID = 4806179088639914364L;
+
+  protected final Throwable exception;
+  
+  // Indicates which statement results in the given exception. 
+  protected final int statementIndex;
+
+  public ExceptionCheck(Throwable exception, int statementIndex) {
+    this.exception = exception;
+    this.statementIndex = statementIndex;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
In that case, I would add a brief comment so stating.
(PS: when replying to a GitHub pull request discussion by email, there is no need to quote the message.)